_posts/2017-11-15-containerised-application-deployment-solutions.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: Deployment solutions of a containerised application
+date: 2017-11-15
+---
+
+I have spent part of last week pulling my hair out over container deployments. If you have read my previous posts, you know by now that I have developed a very simple application and packaged into a Docker container. I am currently trying to deploy this application and I never thought deployment would be the challenging part. I looked at container deployments offered by cloud providers because I thought it would be the easiest and cheapest solutions. It probably is the cheapest but not necessarily the easiest!
+
+In this post I am sharing my experience with four cloud providers that offer container deployment service. If you are reading this article, bear in mind that I am no expert in cloud deployments and I am looking at a very particular use case: deployment of a Docker container that needs good availability (it is okay if it goes down once in a while) and does not need to be scalable (one instance running is enough).
+
+#### Search criteria
+
+Obviously, the solution needs to cover my use case, i.e. deployment of a web application packaged into a Docker container. On top of that here are my criterias:
+
+* **Price**
+<br/>The project I am working on is just a pet project and I won't be making any money with it, so the cheaper, the better.
+
+* **Configuration**
+<br/>I am after a solution easy to configure. My intention is not to become a cloud deployment expert and spend ages learning how to use a particular deployment service, so the easier to configure and learn the better.
+
+* **One-click deployment**
+<br/>The deployment itself must be easy and provides an API or a way for me to easily integrate it with my continuous integration pipeline.
+
+
+#### Microsoft Azure
+
+[Microsoft Azure](https://azure.microsoft.com/) offers very good [accounts](https://azure.microsoft.com/en-gb/free/free-account-faq/) that allow users to try many different services for free.
+In terms of container deployments it provides a wide range of solutions. It makes things a bit confusing to be honest:
+
+* [Azure Container Service](https://azure.microsoft.com/en-gb/services/container-service/) (AKS)
+<br/>This service seems to be great for complex containerised applications that require load-balancing, easy scaling, availability and container orchestration.
+<br/>In my case, while I want my service to be always available, the rest is not going to be a problem and the complexity this service is adding is not worth it. As an example, it uses [Kubernetes](https://kubernetes.io/) as default orchestrator, do I really need it for my one container application? ^^
+
+* [Container Instance](https://azure.microsoft.com/en-gb/services/container-instances/)
+<br/>This service addresses a different type of needs to AKS. It offers like a computing farm where users can send jobs on-demand and paid for the processing time.
```

Comment:
Need~s~

Reply:
You don't need the 'like', unless you want to sound like a Californian, dude :)

Reply:
^ heh. Also, paid -> pay?

Reply:
👍 
hahaha.
👍